src/model.js

```diff
@@ -1875,18 +1875,8 @@ ${associationOwner._getAssociationDebugList()}`);
     }
 
     options = Utils.cloneDeep(options);
-
-    if (options.limit === undefined) {
```

Comment:
I'm not sure why but keeping this if statement fixes the failing tests.
```js
if (options.limit === undefined) {
  options.limit = 1;
}
```
seems to work just fine

Reply:
One of the failing tests is below.
For some reason the MSSQL query is including the `User.id` column in the query even after it is deleted on line 666 🤘.

Logging the sql string in `abstract/query-interface.js#select`:

```sql
SELECT [id], [title], [createdAt], [updatedAt], [user_name] FROM [Tasks] AS [Task] WHERE [Task].[title] = N'some task' ORDER BY [Task].[id] OFFSET 0 ROWS FETCH NEXT 1 ROWS ONLY;
SELECT [username], [createdAt], [updatedAt] FROM [Users] AS [User] WHERE [User].[username] = N'bob' ORDER BY [User].[id] OFFSET 0 ROWS FETCH NEXT 1 ROWS ONLY;
```

Notice `[User].[id]` in the order statement.

https://github.com/sequelize/sequelize/blob/428a85b1376b09373a362d42abedfae0790b8293/test/integration/associations/belongs-to.test.js#L662-L682

Reply:
Dug a bit deeper. And I think I found the source of the issue. 

`src/dialects/mssql/query-generator.js#addLimitAndOffeset`.

`primaryKeyField` is used to order the subqueries. You'll notice in my comment above that the primary key would have been deleted.

See the following console output.

```
console.log('shouldn't be', Object.keys(model.rawAttributes).join(' 🤝 '));
shouldn't be username 🤝 createdAt 🤝 updatedAt
console.log('still here','${this.quoteTable(options.tableAs || model.name)}.${this.quoteIdentifier(model.primaryKeyField)}'
still here [User].[id]
'😭 ${model.primaryKeyField}'
😭 id
```

Not quite sure what the resolution is here. If the PK has been deleted.....